### ISSUE TITLE
Can not install nightly version with unofficial mirrors

### ISSUE THREAD

**Issue by melonedo** [johnnychen94/jill.py#70]
It seems that currently most julia release mirrors do not host nightly builds. As a result, `jill install nightly` errors if I am not using the official mirror.

```
PS D:\code> jill install nightly --upstream SJTUG
JILL - Julia Installer 4 Linux (MacOS, Windows and FreeBSD) -- Light

jill will:
  1) install Julia latest for winnt-x86_64 into C:\Users\melonedo\AppData\Local\julias
  2) make symlinks in C:\Users\melonedo\AppData\Local\julias\bin
You may need to manually add C:\Users\melonedo\AppData\Local\julias\bin to PATH
Continue installation? [Y/n]
----- Download Julia -----
downloading Julia release for latest-winnt-x86_64
failed to find latest-winnt-x86_64 in available upstream. Please try it later.
False
```

**Comment by johnnychen94** [comment]
Correct, those mirror sites don't mirror nightlies for practical reasons.

We could probably use `Official` as a fallback solution by retry with `registry = SourceRegistry(upstream='Official')`

https://github.com/johnnychen94/jill.py/blob/40b290a2fc446fde4641608d17aae9e96a98b33f/jill/download.py#L145-L151

Wanna make a PR?

**Comment by melonedo** [comment]
Thanks for your reply! I will make a pr then.

**Comment by johnnychen94** [comment]
this should be already fixed by #72
